lldb/CMakeLists.txt

```diff
@@ -38,6 +38,12 @@ endif()
 include(LLDBConfig)
 include(AddLLDB)
 
+# This has been added to keep the AIX build isolated for now.
+# It will need to be modified later.
+if (UNIX AND ${CMAKE_SYSTEM_NAME} MATCHES "AIX")
+  add_definitions("-D__AIX__")
```

Comment:
We shouldn't be defining a reserved identifier. Also, isn't there an existing macro that we could test use to check if we're on AIX?

Reply:
It would be really helpful, if we can decide on a better alternative.

Reply:
You didn't really answer my question. Are there predefined compiler macros that we could use? I'd be surprised if there weren't any, as every platform I know of has some. FWIW, these are the predefined macros on linux: https://godbolt.org/z/EG9fKaxMv

Reply:
At this point we are not aware of any predefined ones for AIX though. It would be great if the community can suggest something in this regard. We will also try to find some alternative accordingly.

Reply:
There is `_AIX` at least in clang: https://godbolt.org/z/95MzjcsKn

And according to this (totally official, legit) doc (https://www.lnf.infn.it/computing/doc/aixcxx/html/language/ref/rnmcradd.htm) they exist for the AIX system compiler as well. I'm sure you can find the proper page :)

https://github.com/llvm/llvm-project/blob/0fa78b6c7bd43c2498700a98c47a02cf4fd06388/clang/lib/Basic/Targets/OSTargets.h#L641

If for some reason that's not going to work, then we would use something like `LLDB_...` as we have done for `LLDB_ENABLE_LIBXML2` and friends.

(but I'm yet to look at the whole PR so I don't know yet)

Reply:
Ok Thats helpful 😄 
I will give it go, trying to integrate it with the overall changes. 
Meanwhile, shall I remove `__AIX__` from this PR to keep that discussion separate?

Reply:
Yes. If we need a cmake generated definition, add that in its own PR along with anything that uses it.

Reply:
Well, with *something* that uses it. I'm not asking you to look into the future and know all possible uses.